### ISSUE TITLE
Emit an event before view render

### ISSUE THREAD

**Issue by dan-omniscience** [balderdashy/sails#2747]
Hi,

I been trying to create dynamic menu that needs to be reconstructed regularly so I decided to add an event before the render view call 
https://github.com/dan-omniscience/sails/commit/2ddd96dd0d3492a90228e7a4ab4b2a44fbc03797#diff-9be007416a8b2bcb7f3e7f265ec7fda1R236

I wanted to hear if the community and the core team like this feature and wish to merge this or have any suggestion on a better way to implements this.

Thanks

**Comment by tjwebb** [comment]
What logic is running inside the handler for this event? I'm not sure why this logic couldn't just be in a Controller method

**Comment by dan-omniscience** [comment]
When you set the logic in a controller you can set template data only for
that controller action, this event may be used in a case you will like to
set some data for a template from a hook and not in a specific action.

Dan Aharon-Shalom

**Comment by jdcauley** [comment]
@tjwebb is adding a hook something you would like to do?

**Comment by sailsbot** [comment]
Thanks for posting, @dan-omniscience.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
